/* Copyright 2019 The Brave Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at http://mozilla.org/MPL/2.0/. */

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package brave_pyxis;

message PyxisValue {
  bytes ciphertext = 1;
  bytes tag = 2;
  bytes nonce = 3;
  fixed64 metric_id = 4;

  bytes client_public_key = 5;
}

message RawP3AValue {
  fixed64 metric_id = 1;
  bytes p3a_info = 2;
}

message PyxisMessage {
  repeated PyxisValue pyxis_values = 1;
}
